sdk/include/FreeRTOS-Compat/semphr.h

```diff
@@ -44,7 +169,7 @@ __always_inline static inline SemaphoreHandle_t
 xSemaphoreCreateCounting(UBaseType_t uxMaxCount, UBaseType_t uxInitialCount)
 {
 	SemaphoreHandle_t semaphore =
-	  (SemaphoreHandle_t)malloc(sizeof(struct CountingSemaphoreState));
+	  (SemaphoreHandle_t)malloc(sizeof(SemaphoreHandle_t));
```

Comment:
Do you really mean `sizeof(SemaphoreHandle_t)`?

```suggestion
	  (SemaphoreHandle_t)malloc(sizeof(StaticSemaphore_t));
```